### PR TITLE
[DEP] Move HampelFilter to outliers

### PR DESCRIPTION
**Pull request by TonyBagnall**
Deprecates the HampelFilter transformer and moves functionality to anomaly detection, as discussed on slack

part of #1446

### CODE REVIEW

File: aeon/transformations/outlier_detection.py [hunk 1]
```diff
@@ -7,11 +7,19 @@
...
 
 from aeon.forecasting.model_selection import SlidingWindowSplitter
 from aeon.transformations.base import BaseTransformer
 
 
+# TODO: remove v0.10.0
+@deprecated(
+    version="0.9.0",
+    reason="The HampelFilter transformer will be removed in v0.10.0 and will be "
+    "replaced by an anomaly detector with ",
```
Comment (CodeLionX) [inline aeon/transformations/outlier_detection.py:20]: with what?

File: aeon/anomaly_detection/_hampel.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    >>> y_hat = transformer.fit_transform(y)
+    """
+
+    _tags = {
+        "input_data_type": "Series",
+        "output_data_type": "Series",
+        "X_inner_type": ["pd.DataFrame", "pd.Series"],
+        "capability:missing_values": True,
+        "capability:multivariate": True,
+    }
```
Comment (MatthewMiddlehurst) [inline aeon/anomaly_detection/_hampel.py:52]: These are outdated now.

File: aeon/anomaly_detection/_hampel.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        ----------
+        X : pd.Series or pd.DataFrame
+            Data to be transformed
+        y : ignored argument for interface compatibility
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        Xt : pd.Series or pd.DataFrame, same type as X
+            transformed version of X
```
Comment (MatthewMiddlehurst) [inline aeon/anomaly_detection/_hampel.py:75]: Same here, a lot of the documentation does not seem to fit with the move.

File: aeon/anomaly_detection/_hampel.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        """
+        Z = X.copy()
+
+        # multivariate
+        if isinstance(Z, pd.DataFrame):
+            for col in Z:
+                Z[col] = self._predict_outliers(Z[col])
```
Comment (MatthewMiddlehurst) [inline aeon/anomaly_detection/_hampel.py:82]: I think testing may not be working as intended if this is passing

**Pull request closed without merging**